### PR TITLE
New Red Alert Squad: NT Flamers!

### PR DESCRIPTION
**Pull request by ghost**
:cl: Armhulen for code, Dunham for sprites and ShadowDimentio as the ideas guy
add: Leaked documents of a secret team of elite nanotrasen firebugs has been spread around this side of the milky way, and citizens are quite abuzz in fear! They're probably not REAL, though. That would be crazy.
/:cl:

What's cooler than shooting zombies? Burning them! Blobs and kudzu will fear these cold, heartless monsters when they arrive on the station. They are the Incinerator Team! They have infinite flamethrowers (they regenerate ammo over time, cremating regenerates the ammo) and they may switch their flamethrower to cremate mode to suck up corpses to burn.

image

(for some reason chairs won't load >:( )

### CODE REVIEW

**Comment by Hatterhat** [comment]
that's a fuckin hot n spicy meme @Armhulen

**Comment by Armhulen** [comment]
:)

**Comment by Armhulen** [comment]
so you be sayin it compiles but when travis sees it...

**Comment by kevinz000** [comment]
mfw arm doesn't know that admins can already make custom sprited infinite flamethrower deathsquad......

File: code/modules/admin/verbs/one_click_antag.dm [hunk 1]
```diff
@@ -357,6 +357,71 @@
...
+			chosen_candidate.client.prefs.copy_to(Incinerator)
+			if(numagents == 1) //If Squad Leader
+				Incinerator.equipOutfit(/datum/outfit/death_commando/officer)
+			else
+				Incinerator.equipOutfit(/datum/outfit/death_commando)
+			Incinerator.dna.update_dna_identity()
+			Incinerator.key = chosen_candidate.key
+			Incinerator.mind.assigned_role = "Death Commando"
+
+			//Assign antag status and the mission
```
Comment (kevinz000) [inline code/modules/admin/verbs/one_click_antag.dm]: nanotrasen sending antags *sigh*

Comment (Armhulen) [reply code/modules/admin/verbs/one_click_antag.dm]: nothing personell

**Comment by Armhulen** [comment]
@kevinz000 explain, NOW.

File: code/modules/admin/verbs/one_click_antag.dm [hunk 2]
```diff
@@ -357,6 +357,71 @@
 
 	return
 
+// INCINERATORS
```
Comment (JJRcop) [inline code/modules/admin/verbs/one_click_antag.dm]: This is still in the madeDeathsquad proc, you forgot to define makeIncinerators

File: code/modules/admin/verbs/one_click_antag.dm [hunk 3]
```diff
@@ -357,6 +357,71 @@
...
+			if(numagents == 1)
+				message_admins("The Incineraton team has spawned with the mission: [mission].")
+			log_game("[key_name(Incinerator)] has been selected as an Incinerator")
+			numagents--
+
+			return 1
+		else
+			return 0
+
+	return
```
Comment (JJRcop) [inline code/modules/admin/verbs/one_click_antag.dm]: trailing return, remove

**Comment by Armhulen** [comment]
i fixed the incinerators thing, just didn't sync. One second!

File: code/modules/admin/verbs/one_click_antag.dm [hunk 4]
```diff
@@ -357,6 +357,71 @@
...
+
+			if(CONFIG_GET(flag/enforce_human_authority))
+				Incinerator.set_species(/datum/species/human)
+
+			//Logging and cleanup
+			if(numagents == 1)
+				message_admins("The Incineraton team has spawned with the mission: [mission].")
+			log_game("[key_name(Incinerator)] has been selected as an Incinerator")
+			numagents--
+
```
Comment (JJRcop) [inline code/modules/admin/verbs/one_click_antag.dm]: missing ```if(packSpawned)```

**Comment by JJRcop** [comment]
Did you forget to commit some files too?

File: code/modules/admin/verbs/one_click_antag.dm (hunk 2 above)
Comment (JJRcop) [reply code/modules/admin/verbs/one_click_antag.dm]: resolved

**Comment by JJRcop** [comment]
Since the makeIncinerators proc is so similar to the makeDeathsquad proc I feel you can merge them together, maybe even make a datum """antag""" for both deathsquaddies and incinerators and give the objective text there.

**Comment by JJRcop** [comment]
Maybe it's not necessary to use datum/antagonist but some form of datumization is optimal here

**Comment by Armhulen** [comment]
pass, that sounds like a really big headache. Sorry

File: code/modules/admin/verbs/one_click_antag.dm (hunk 1 above)
Comment (Supermichael777) [reply code/modules/admin/verbs/one_click_antag.dm]: NANI!?

**Comment by WJohn** [comment]
Those sprites look great!

**Comment by Steelpoint** [comment]
The sprites look like the Head of Security found a orange spray can. You'd be better off getting the default ERT sprites and making a more dark red version. 

Incinerator ERT seems redundant in my opinion, the current ERT function well enough and have scaling power at the discretion of the admins, and the Deathsquad just kill everything.

File: code/modules/clothing/outfits/standard.dm [hunk 5]
```diff
@@ -392,6 +392,53 @@
...
+	R.freqlock = 1
+
+	var/obj/item/implant/mindshield/L = new/obj/item/implant/mindshield(H)//Here you go Deuryn
+	L.implant(H, null, 1)
+
+
+	var/obj/item/card/id/W = H.wear_id
+	W.icon_state = "centcom"
+	W.access = get_all_accesses()//They get full station access.
+	W.access += get_centcom_access("NT Flamer")//Let's add their alloted CentCom access.
```
Comment (RemieRichards) [inline code/modules/clothing/outfits/standard.dm]: nice non-functional copypasta

**Comment by Armhulen** [comment]
@Steelpoint the commander has new sprites now, and besides the art lead disagrees with you soooo

File: code/modules/clothing/outfits/standard.dm (hunk 5 above)
Comment (Armhulen) [reply code/modules/clothing/outfits/standard.dm]: i'm only human after all, don't put the blame on me, no don't put the blame on me

(do i just get_centcom_access from the commando or)

Comment (RemieRichards) [reply code/modules/clothing/outfits/standard.dm]: there is no "NT Flamer" option in the switch in ```get_centcom_access()```
either:
- Add it, and its appropriate accesses
or
- Reuse an existing type, like the commando.

Comment (Armhulen) [reply code/modules/clothing/outfits/standard.dm]: okay, thank you!

**Comment by Armhulen** [comment]
What's left, my friends?
* grab flamethrower code from kevinz
* add in the sprites and stuff for ^^
* fix flamer access on centcom
and i THINK that's it!

File: code/modules/clothing/spacesuits/miscellaneous.dm [hunk 6]
```diff
@@ -42,6 +43,54 @@ Contains:
 	helmettype = /obj/item/clothing/head/helmet/space/hardsuit/deathsquad
 	dog_fashion = /datum/dog_fashion/back/deathsquad
 
+/obj/item/clothing/head/helmet/space/hardsuit/flamer
+	name = "MK.II FLAME Helmet"
+	desc = "A space proof, fixium-reinforced SWAT mask. The extra reinforcements means it has to be a helmet, though."
```
Comment (Armhulen) [inline code/modules/clothing/spacesuits/miscellaneous.dm]: by the way, this lore is correct. Fixium is in the description of the bomb test site indestructable wall

**Comment by uraniummeltdown** [comment]
The sprites are cool but other than that it's just "dude what if deathsquad/ERT was fire"

**Comment by KingPhilipIII** [comment]
@uraniummeltdown You say that like it's a bad thing.

**Comment by Armhulen** [comment]
>not sending firesquad to burn away the kudzu

>not replacing the boring you took too long to win nuke in blob gamemode with squads of incinerators to fight the blob

ok

**Comment by Bawhoppen** [comment]
Yea I give arm about a month before he burns out and never comes back

**Comment by Armhulen** [comment]
Jut going down the checklist

**Review by ShizCalev — CHANGES_REQUESTED** [review]
Need to run mapmerge on CentCom.dmm

**Comment by DoctoreMad** [comment]
>and they may switch their flamethrower to cremate mode to suck up corpses to burn.

Brutal.

**Comment by Armhulen** [comment]
i dunno i'm waiting on kev to finish the flamethrower code or whatever

**Comment by Cyberboss** [comment]
Closed for baiting me

**Comment by duncathan** [comment]
gbpssssss

**Pull request closed without merging**